lib/HLSL/HLOperationLower.cpp

```diff
@@ -5424,7 +5400,7 @@ Value *TranslateConstBufMatLdLegacy(Type *matType, Value *handle,
                                     IRBuilder<> &Builder) {
   unsigned col, row;
   HLMatrixLower::GetMatrixInfo(matType, col, row);
-  Type *EltTy = HLMatrixLower::LowerMatrixType(matType, /*forMem*/memElemRepr)->getVectorElementType();
+  Type *EltTy = HLMatrixLower::LowerMatrixType(matType, /*forMem*/true)->getVectorElementType();
```

Comment:
We should always load using the mem representation, but the epilogue will convert to the register representation if that's what the caller wants.